### PR TITLE
Added subscription mentions, edits and deletes.

### LINKED ISSUE #36: Subscription edits and deletes
**Please describe what you'd like the bot to do**
When a message is deleted is edited from a subscription based channel, it should also be edited or deleted in the subscriber's channel. This means there aren't any silly typos or duplicated messages sent.

**Why should the bot do this?**
When a message is edited or deleted, it's usually because of some mistake, it would be good to make it so that those mistakes are less visable.

**Additional context**

### PR DESCRIPTION
**Pull request by Greenfoot5**
Fixes #36 and #37.

### CODE REVIEW

File: cogs/subscriptions.py [hunk 1]
```diff
@@ -244,6 +252,98 @@ async def on_message(self, ctx):
                 except discord.Forbidden:
                     pass
 
+    @commands.Cog.listener()
+    async def on_message_edit(self, before, after):
+        # Obtains the relebent data.
+        subs = data_handler.load("subscriptions")
+
+        # Stops a chain response.
+        if self.bot.user.id == before.author.id:
```
Comment (Q-Sharp) [inline cogs/subscriptions.py:258]: No need to first load the json data before.

File: cogs/subscriptions.py [hunk 2]
```diff
@@ -244,6 +252,98 @@ async def on_message(self, ctx):
...
+
+                async for mesg in channel.history(limit = 200, after = dayAgo):
+                    if before.content in mesg.content and mesg.author.id == self.bot.user.id:
+                        msg = mesg
+                # Convert the message so we can output it all correctly.
+                # Checks if it's a system message, if so, output it nicely.
+                if after.is_system():
+                    message = f">>> {after.system_content}"
+                # Checks if the content is empty, if no, just add the author's name.
+                elif after.content == "":
```
Comment (Q-Sharp) [inline cogs/subscriptions.py:287]: " " <--- wouldn't this be empty, too?

File: cogs/subscriptions.py [hunk 3]
```diff
@@ -244,6 +252,98 @@ async def on_message(self, ctx):
...
+                    await msg.delete()
+                # If the bot doesn't have permissions, just ignore it.
+                except discord.Forbidden:
+                    pass
+
+
+
 # Setup the cog correctly when called to do so.
 def setup(bot):
     bot.add_cog(Subscriptions(bot))
```
Comment (Q-Sharp) [inline cogs/subscriptions.py]: https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

**Review by Q-Sharp — COMMENTED** [review]
I added some comments in source. All in all it seems to be good work :)

File: cogs/subscriptions.py (hunk 2 above)
Comment (Greenfoot5) [reply cogs/subscriptions.py:287]: Empty is "" I think You can't have a message with a content of " ". `None` could also work. I'll have to check this on the d.py server.

Comment (Greenfoot5) [reply cogs/subscriptions.py:287]: Using "" is correct, as that's an empty string.

**Comment by Greenfoot5** [comment]
Fixed that all, wanna give it the once over then merge?

**Review by Q-Sharp — APPROVED** [review]

**Pull request merged**